### PR TITLE
BAU: Add local startup configuration

### PR DESCRIPTION
**Pull request by CharlesIC**
This adds the ability to easily run the Proxy Node apps locally with minimal set-up using Docker. There are certain limitations to this at the moment, described in the instructions.

Recommend reviewing the `local-startup/running-proxy-node-locally.md` file first.

  - Provide Dockerfiles for local running
  - Provide `docker.env` file to set up apps for local running
  - Add `docker-compose.yml` file to run Proxy Node apps easily
  - Add script to start up Proxy Node
  - Add local running instructions
  - Update gradle to latest version
  - Add metadata files - these have been manually signed with `xmlsectool` and cannot be changed without re-signing
  - Add metadata truststores - these have been generated manually with `keytool` and need to be re-generated if metadata gets re-signed with a different cert chain
  - Add local PKI (keys and certs symlinked from `verify-dev-pki`)
  - Use locally built artefacts in Docker images by default 
  - Pull VSP image from Proxy Node registry rather than building it from scratch

To test a local journey run `./docker-startup.sh --build` and open `http://localhost:6610/` in your browser. See README for instructions on how to deal with the Compliance Tool.

### CODE REVIEW

File: local-startup/Dockerfile
Comment (rhowe-gds) [inline local-startup/Dockerfile]: recommend `["bin/$COMPONENT"]` to avoid wrapping it in `sh -c`

File: local-startup/jdk11-jre-image/Dockerfile
Comment (rhowe-gds) [inline local-startup/jdk11-jre-image/Dockerfile]: same `CMD` style recommendation as above

File: local-startup/running-proxy-node-locally.md
Comment (rhowe-gds) [inline local-startup/running-proxy-node-locally.md]: If `USE_LOCAL_BUILD` only applies to `docker-compose`, you can run it as

`USE_LOCAL_BUILD=true docker-compose ...` which will avoid polluting your shell environment

File: proxy-node-vsp-config/verify-service-provider.yml
Comment (rhowe-gds) [inline proxy-node-vsp-config/verify-service-provider.yml]: Will the service work if this is blank? If not, why even have the fallback? Have dropwizard throw an error due to a missing interpolation token instead.

File: docker-compose.yml
Comment (tomrosier) [inline docker-compose.yml]: Is it worth switching theses to 6*** port just to be inline with the other ones.

File: local-startup/running-proxy-node-locally.md
Comment (tomrosier) [inline local-startup/running-proxy-node-locally.md]: We should discuss the SoftHSM stuff, as it wasn't appropriate when we tried to use it before as its implementation is vastly different from the AWS HSM implementation.

File: startup-docker.sh
Comment (tomrosier) [inline startup-docker.sh]: Is this redundant?

**Review by tommrr — COMMENTED** [review]
LGTM, will approve after Russells changes have been applied

**Comment by rhowe-gds** [comment]
> LGTM, will approve after Russells changes have been applied

Don't feel you need to hold off on my comments - they're merely suggestions and I don't think there'll be any real change in behaviour

File: docker-compose.yml
Comment (CharlesIC) [reply docker-compose.yml]: Not really - VSP isn't our app. It's fine if it has a different port number.

File: local-startup/running-proxy-node-locally.md
Comment (CharlesIC) [reply local-startup/running-proxy-node-locally.md]: Yeah we might not be able to use it. I've not looked into it in much detail but we should definitely look into it.

File: startup-docker.sh
Comment (CharlesIC) [reply startup-docker.sh]: Nope - because above we use `printf` so there's no newline at the end

File: local-startup/Dockerfile
Comment (CharlesIC) [reply local-startup/Dockerfile]: We need variable substitution, though. [Docker docs](https://docs.docker.com/engine/reference/builder/#cmd) say "if you want shell processing then either use the shell form or execute a shell directly". So it's got to be the _exec_ form or this:
`CMD ["/bin/sh", "-c", "\"bin/${COMPONENT}\""]`

**Review by JohnWattsGDS — APPROVED** [review]
This is probably fine, if it doesn't work, I'll find out when I try to run it locally.

**Comment by CharlesIC** [comment]
Simply run `./startup-docker.sh --build` - it's magic! 🎩✨

**Review by idrop — APPROVED** [review]
🚀

**Pull request merged**
